bigtable/admin.go

```diff
@@ -2150,9 +2150,60 @@ func (ac *AdminClient) RestoreTableFrom(ctx context.Context, sourceInstance, tab
 	return longrunning.InternalNewOperation(ac.lroClient, op).Wait(ctx, &resp)
 }
 
+type backupOptions struct {
+	backupType        *BackupType
+	hotToStandardTime *time.Time
+}
+
+// BackupOption can be used to specify parameters for backup operations.
+type BackupOption interface {
+	apply(o *backupOptions)
+}
+
+// setHotToStandardTime can be used to set backup type to [BackupTypeHot] and the
+// time at which the hot backup will
+// be converted to a standard backup.
+// Once the `hot_to_standard_time` has passed, Cloud Bigtable will convert the
+// hot backup to a standard backup. This value must be greater than the backup
+// creation time by at least 24 hours.
+type hotBackupOption struct {
+	htsTime *time.Time
+}
+
+func (hbo hotBackupOption) apply(o *backupOptions) {
+	o.hotToStandardTime = hbo.htsTime
+	btHot := BackupTypeHot
+	o.backupType = &btHot
+}
+
+// Ptr returns a pointer to its argument.
+// It can be used to initialize pointer fields:
+//
+//	E.g. HotBackup(bigtable.Ptr[time.Time](time.Now()))
+func Ptr[T any](t T) *T { return &t }
+
+// HotBackup can be used to set backup type to [BackupTypeHot] and optionally, specify the
+// time at which the hot backup will be converted to a standard backup.
+// Once the `hot_to_standard_time` has passed, Cloud Bigtable will convert the
+// hot backup to a standard backup. This value, if provided, must be greater than the backup
+// creation time by:
+// - At least 24 hours
+func HotBackup(hotToStandardTime *time.Time) BackupOption {
+	return hotBackupOption{htsTime: hotToStandardTime}
+}
```

Comment:
Why is this option provided as a pointer instead of passing in a value then converting it into a reference in the library?

Reply:
hotToStandardTime is optional here same as the protos:

https://github.com/googleapis/google-cloud-go/blob/649c075d5310e2fac64a0b65ec445e7caef42cb0/bigtable/admin/apiv2/adminpb/table.pb.go#L1390-L1400

Reply:
I've been trying to also come up with ways to represent an optional `Time` parameter here, even looking at https://stackoverflow.com/questions/30716354/how-do-i-do-a-literal-int64-in-go for help. You could do something like `HotBackup(time.Time)` then another backup option like `HotBackupNil()` or something like that. Not sure if it's the cleanest solution to this problem, but would also like to hear from others.

Reply:
Changed the options to `HotToStandardBackup` and `HotBackup`
Open to other naming suggestions

Reply:
I'd leave the final decision for the names to the Bigtable team, but setting `hot_to_standard_time` in the proto means setting the time at which it gets converted, and if doing a hot backup without having a provided value means it doesn't get converted, right?

In that case I'd rename the two to something like `HotBackupWithConversion(hotToStandardTime time.Time)` and `HotBackupWithoutConversion()`, as it's more clear to the user what those options are doing.